scripts/ci/get_python_version.py

```diff
@@ -0,0 +1,132 @@
+#!/usr/bin/env python
+
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+from subprocess import check_output
+from util import SRC_PATH
+import json
+import logging
+import os
+import shlex
+import subprocess
+import sys
+
+logger = logging.getLogger(__name__)
+logger.setLevel(logging.DEBUG)
+ch = logging.StreamHandler()
+ch.setLevel(logging.DEBUG)
+logger.addHandler(ch)
+
+
+def get_all_tests():
+    all_tests = []
+    for src_d in os.listdir(SRC_PATH):
+        src_d_full = os.path.join(SRC_PATH, src_d)
+        if not os.path.isdir(src_d_full):
+            continue
+        pkg_name = next((d for d in os.listdir(src_d_full) if d.startswith('azext_')), None)
+
+        # If running in Travis CI, only run tests for edited extensions
+        commit_range = os.environ.get('TRAVIS_COMMIT_RANGE')
+        if commit_range and not check_output(
+                ['git', '--no-pager', 'diff', '--name-only', commit_range, '--', src_d_full]):
+            continue
+
+        # Running in Azure DevOps
+        cmd_tpl = 'git --no-pager diff --name-only origin/{commit_start} {commit_end} -- {code_dir}'
+        ado_branch_last_commit = os.environ.get('ADO_PULL_REQUEST_LATEST_COMMIT')
+        ado_target_branch = os.environ.get('ADO_PULL_REQUEST_TARGET_BRANCH')
+        if ado_branch_last_commit and ado_target_branch:
+            if ado_branch_last_commit == '$(System.PullRequest.SourceCommitId)':
+                # default value if ADO_PULL_REQUEST_LATEST_COMMIT not set in ADO
+                continue
+            elif ado_target_branch == '$(System.PullRequest.TargetBranch)':
+                # default value if ADO_PULL_REQUEST_TARGET_BRANCH not set in ADO
+                continue
+            else:
+                cmd = cmd_tpl.format(commit_start=ado_target_branch, commit_end=ado_branch_last_commit,
+                                     code_dir=src_d_full)
+                if not check_output(shlex.split(cmd)):
+                    continue
+
+        # Find the package and check it has tests
+        test_dir = os.path.isdir(os.path.join(src_d_full, pkg_name, 'tests'))
+        if pkg_name and test_dir:
+            # [('azext_healthcareapis', '/mnt/vss/_work/1/s/src/healthcareapis')]
+            all_tests.append((pkg_name, src_d_full))
+            # azext_healthcareapis -> healthcareapis
+            _, extension_name = pkg_name.split('_', 1)
+            logger.info(f'ado_branch_last_commit: {ado_branch_last_commit}, '
+                        f'ado_target_branch: {ado_target_branch}, '
+                        f'detect which extension need to test: {all_tests}.')
+            return pkg_name, extension_name
```

Comment:
Should we ignore the scenario in which more than one package is changed?

Reply:
What is taken now is to end immediately when one extension is found. 
Generally speaking, a PR will not contain the code of two extensions at the same time.